core/base/src/TSystem.cxx

```diff
@@ -3374,8 +3380,6 @@ int TSystem::CompileMacro(const char *filename, Option_t *opt,
          if (!keep) k->SetBit(kMustCleanup);
          fCompiled->Add(k);
 
-         gInterpreter->GetSharedLibDeps(library);
```

Comment:
This call is still need for the not-explicitly-linked case.  It has the side effect of load the dependencies from the rootmap files so that the subsequent LoadLibrary can use that information. See commit 8fdd5c3c38892582f29861b18b22ef51bb266375